core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java

```diff
@@ -6713,6 +6713,57 @@ private void checkNullOperand(SqlTester tester, String op) {
         "extract(microsecond from cast(null as time))");
   }
 
+  @Test public void testExtractWithDatesBeforeUnixEpoch() {
+    tester.checkScalar(
```

Comment:
Can you pls also add tests for other TimeUnits like :
* DOW
* DOY
* DECADE
* CENTURY

Reply:
will do